tests/deployment_test.go

```diff
@@ -24,10 +24,269 @@
+			Namespace: d.Namespace(),
+			TaskQueue: taskQueue,
+			Identity:  "random",
+			WorkerVersionCapabilities: &commonpb.WorkerVersionCapabilities{
+				UseVersioning:        true,
+				BuildId:              buildID,
+				DeploymentSeriesName: seriesName,
+			},
+		})
+		select {
+		case <-ctx.Done():
+			return
+		case errChan <- err:
+		}
+	}()
+
+	// Querying the Deployment
+	d.EventuallyWithT(func(t *assert.CollectT) {
+		a := assert.New(t)
+
+		resp, err := d.FrontendClient().GetCurrentDeployment(ctx, &workflowservice.GetCurrentDeploymentRequest{
+			Namespace:  d.Namespace(),
+			SeriesName: seriesName,
+		})
+		a.NoError(err)
+		a.Nil(resp.GetCurrentDeploymentInfo())
+
+	}, time.Second*5, time.Millisecond*200)
+
+	<-ctx.Done()
+	select {
+	case err := <-errChan:
+		d.Fail("Expected error channel to be empty but got error %w", err)
+	default:
+	}
+}
+
+func (d *DeploymentSuite) listDeployments(seriesName string, buildID string, withSeries bool) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	// todo (shivam) - fails when seriesName has backslashes
+	taskQueue := &taskqueuepb.TaskQueue{Name: "deployment-test", Kind: enumspb.TASK_QUEUE_KIND_NORMAL}
+	numberOfDeployments := 1
+
+	// Starting a deployment workflow
+	errChan := make(chan error)
+	defer close(errChan)
+
+	go func() {
+		_, err := d.FrontendClient().PollWorkflowTaskQueue(ctx, &workflowservice.PollWorkflowTaskQueueRequest{
+			Namespace: d.Namespace(),
+			TaskQueue: taskQueue,
+			Identity:  "random",
+			WorkerVersionCapabilities: &commonpb.WorkerVersionCapabilities{
+				UseVersioning:        true,
+				BuildId:              buildID,
+				DeploymentSeriesName: seriesName,
+			},
+		})
+		select {
+		case <-ctx.Done():
+			return
+		case errChan <- err:
+		}
+	}()
+
+	request := &workflowservice.ListDeploymentsRequest{
+		Namespace: d.Namespace(),
+	}
+	if withSeries {
+		request.SeriesName = seriesName
+	}
+
+	// Querying the Deployment
+	d.EventuallyWithT(func(t *assert.CollectT) {
+		a := assert.New(t)
+
+		resp, err := d.FrontendClient().ListDeployments(ctx, request)
+		a.NoError(err)
+		a.NotNil(resp)
+		if resp == nil {
+			return
+		}
+		a.NotNil(resp.Deployments)
+		a.Equal(numberOfDeployments, len(resp.Deployments))
+		if len(resp.Deployments) < 1 {
+			return
+		}
+
+		deployment := resp.Deployments[0]
+		a.Equal(seriesName, deployment.Deployment.SeriesName)
+		a.Equal(buildID, deployment.Deployment.BuildId)
+		a.Equal(false, deployment.IsCurrent)
+
+	}, time.Second*5, time.Millisecond*200)
+
+	<-ctx.Done()
+	select {
+	case err := <-errChan:
+		d.Fail("Expected error channel to be empty but got error %w", err)
+	default:
+	}
+}
+func (d *DeploymentSuite) TestListDeployments_WithSeriesName() {
+	seriesName := "abc"
+	buildID := "xyz"
+	d.listDeployments(seriesName, buildID, true)
+}
+
+func (d *DeploymentSuite) TestListDeployments_WithoutSeriesName() {
+	seriesName := "d"
```

Comment:
Using different names for series + build since our cluster will already have a deployment entity wf running when the previous test runs - making the assertions to fail

Reply:
RandomizeStr would help with this, although based on the test name (if I understand it correctly) you don't want to pass a series filter.